fetch-nixpkgs.nix

```diff
@@ -1,23 +1,8 @@
 let
-  spec = builtins.fromJSON (builtins.readFile ./nixpkgs-src.json);
-  src = import <nix/fetchurl.nix> {
-    url = "https://github.com/${spec.owner}/${spec.repo}/archive/${spec.rev}.tar.gz";
-    inherit (spec) sha256;
-  };
-  nixcfg = import <nix/config.nix>;
-in builtins.derivation {
-  system = builtins.currentSystem;
-  name = "${src.name}-unpacked";
-  builder = builtins.storePath nixcfg.shell;
-  inherit src;
-  args = [
-    (builtins.toFile "builder" ''
-      $coreutils/mkdir $out
-      cd $out
-      $gzip -d < $src | $tar -x --strip-components=1
-    '')
-  ];
-  coreutils = builtins.storePath nixcfg.coreutils;
-  tar = builtins.storePath nixcfg.tar;
-  gzip = builtins.storePath nixcfg.gzip;
-}
+  # temporary hack until scripts/nix-shell.sh ceases to use -p
```

Comment:
Can you elaborate what's going on there?

Reply:
`scripts/nix-shell.sh` needs to be able to `nix-build` the nixpkgs to fetch it and populate `NIX_PATH`
but the nix 2.0 `builtins.fetchTarball` doesnt return a derivation, just a string with a storepath, and `nix-build` doesnt like that

so i need a dummy derivation to make `nix-build` happy

once `nix-shell.sh` is redone to not use `-p`, then we dont need this anymore